src/drivers/distance_sensor/hxSRx0x/hxSRx0x_main.cpp

```diff
@@ -0,0 +1,212 @@
+ */
+namespace hxSRx0x
+{
+
+HxSRx0x *instance = nullptr;
+
+int start(const uint8_t rotation = distance_sensor_s::ROTATION_DOWNWARD_FACING);
+int status();
+int stop();
+int usage();
+
+/**
+ * Start the pwm driver.
+ *
+ * This function only returns if the sensor is up and running
+ * or could not be detected successfully.
+ */
+int
+start(const uint8_t rotation)
+{
+	if (instance != nullptr) {
+		PX4_ERR("already started");
+		return PX4_OK;
+	}
+
+	instance = new HxSRx0x(rotation);
+
+	if (instance == nullptr) {
+		PX4_ERR("Failed to instantiate the driver");
+		return PX4_ERROR;
+	}
+
+	// Initialize the sensor.
+	if (instance->init() != PX4_OK) {
+		PX4_ERR("Failed to initialize LidarLite pwm.");
+		delete instance;
+		instance = nullptr;
+		return PX4_ERROR;
+	}
+
+	// Start the driver.
+	instance->start();
+
+	PX4_INFO("driver started");
+	return PX4_OK;
+}
+
+/**
+ * @brief Prints status info about the driver.
+ */
+int
+status()
+{
+	if (instance == nullptr) {
+		PX4_ERR("driver not running");
+		return PX4_ERROR;
+	}
+
+	instance->print_info();
+	return PX4_OK;
+}
+
+/**
+ * @brief Stops the driver
+ */
+int
+stop()
+{
+	if (instance != nullptr) {
+		delete instance;
+		instance = nullptr;
+	}
+
+	return PX4_OK;
+}
+
+/**
+ * @brief Displays driver usage at the console.
+ */
+int
+usage()
+{
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+
+Driver for HY-SRF05 / HC-SR05 and HC-SR04 rangefinders.
+
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("hxSRx0x", "driver");
+	PRINT_MODULE_USAGE_SUBCATEGORY("distance_sensor");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("start","Start driver");
+	PRINT_MODULE_USAGE_PARAM_INT('R', 25, 0, 25, "Sensor rotation - downward facing by default", true);
+	PRINT_MODULE_USAGE_COMMAND_DESCR("status","Print driver status information");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("stop","Stop driver");
+	return PX4_OK;
+}
+
+} // namespace hxSRx0x
+
+
+/**
+ * @brief Driver 'main' command.
+ */
+extern "C" __EXPORT int hxSRx0x_main(int argc, char *argv[])
+{
+	const char *myoptarg = nullptr;
+
+	int ch = 0;
+	int myoptind = 1;
+
+	uint8_t rotation = distance_sensor_s::ROTATION_DOWNWARD_FACING;
+
+	while ((ch = px4_getopt(argc, argv, "R:", &myoptind, &myoptarg)) != EOF) {
+		switch (ch) {
+		case 'R':
+			rotation = (uint8_t)atoi(myoptarg);
+			PX4_INFO("Setting Lidar orientation to %d", (int)rotation);
+			break;
+
+		default:
+			return hxSRx0x::usage();
+		}
+	}
+
+	if (myoptind >= argc) {
+		return hxSRx0x::usage();
+	}
+
+	// Start the driver.
+	if (!strcmp(argv[myoptind], "start")) {
+		return hxSRx0x::start(rotation);
+	}
+
+	// Print the driver status.
+	if (!strcmp(argv[myoptind], "status")) {
+		return hxSRx0x::status();
+	}
+
+	// Stop the driver
+	if (!strcmp(argv[myoptind], "stop")) {
+		return hxSRx0x::stop();
+	}
+
+	// Print driver usage information.
+	return hxSRx0x::usage();
+}
+#else
+extern "C" __EXPORT int hxSRx0x_main(int argc, char *argv[])
+{
+  PX4_ERR("GPIO_ULTRASOUND_xxx not supported.");
```

Comment:
I think it would be better to instead have a #error if the board doesn't provide the required defines, then we'll catch it and simply not have the module included in the build at all.

Reply:
Done